application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -179,6 +185,7 @@ private void processAttachments(@NotNull MessageReceivedEvent event,
 
         if (statusCode < HttpURLConnection.HTTP_OK
                 || statusCode >= HttpURLConnection.HTTP_MULT_CHOICE) {
+            LOGGER.warn("Request JSON: {}", body);
             throw new IllegalStateException("Gist API unexpected response: " + apiResponse.body());
```

Comment:
Please put the body into the exception instead. Remember that the bot is multithreaded, so the warning message can drift away from the exceptions log. then you have something like
```java
WARN Request JSON: ...
INFO dasdasdasdas
WARN dasd
INFO asd
INFO asd
ERROR asd
INFO as
WARN das
INFO das
INFO das
ERROR Gist API unexpected response: ...
```
and nobody can connect the dots anymore. Better to put everything into one message instead of creating multiple messages.